### PR TITLE
Split `SolutionImprovementThresholds` into two types

### LINKED ISSUE #11204: Split `SolutionImprovementThresholds` into two types.
Currently, this type reads: 

https://github.com/paritytech/substrate/blob/07d8c9d50cf5b8f92d04e134b8504167313b8d44/frame/election-provider-multi-phase/src/lib.rs#L587-L590

and the problem is in that *any phase*. For technical reasons beyond the scope of this issue, we want to be able to define `SolutionImprovementThresholds`, with the exact same semantic, for the different `Signed` and `Unsigned` phases. So, we would break this type into two types. 

I think the logic change will be trivial. Some tests should be altered or added to demonstrate what the two configs do. 

Companion should set the `Signed` variant to `Default::default`, and keep the old `SolutionImprovementThresholds` for unsigned.

### PR DESCRIPTION
**Pull request by georgesdib**
Fixes #11204 

As the issue describes, splitting `SolutionImprovementThresholds` into `BetterSignedThreshold` and `BetterUnsignedThreshold`.

Adding some tests to show case the difference.

@kianenigma 

##
Polkadot companion: paritytech/polkadot#5324

##

Polkadot address: 131dPecTmpTC1p1ofemufqFBJo9vNbV7dkgN7vWwKnaSMkC4

### CODE REVIEW

File: bin/node/runtime/src/lib.rs [hunk 1]
```diff
@@ -663,7 +663,8 @@ impl pallet_election_provider_multi_phase::Config for Runtime {
 	type EstimateCallFee = TransactionPayment;
 	type SignedPhase = SignedPhase;
 	type UnsignedPhase = UnsignedPhase;
-	type SolutionImprovementThreshold = SolutionImprovementThreshold;
+	type SolutionImprovementThresholdUnsigned = SolutionImprovementThreshold;
+	type SolutionImprovementThresholdSigned = GetDefault;
```
Comment (kianenigma) [inline bin/node/runtime/src/lib.rs:667]:
```suggestion
	type SolutionImprovementThresholdSigned = ();
```

File: bin/node/runtime/src/lib.rs [hunk 2]
```diff
@@ -26,7 +26,7 @@ use codec::{Decode, Encode, MaxEncodedLen};
 use frame_election_provider_support::{onchain, ExtendedBalance, SequentialPhragmen, VoteWeight};
 use frame_support::{
 	construct_runtime,
-	pallet_prelude::Get,
+	pallet_prelude::{Get, GetDefault},
```
Comment (kianenigma) [inline bin/node/runtime/src/lib.rs:29]:
```suggestion
	pallet_prelude::Get,
```

File: bin/node/runtime/src/lib.rs (hunk 1 above)
Comment (kianenigma) [reply bin/node/runtime/src/lib.rs:667]: Now that we are renaming things, I am a fan of using a shorter name like: `BetterSignedSolution`, `BetterSignedThreshold`, or anything else you can come up with.

Comment (kianenigma) [reply bin/node/runtime/src/lib.rs:667]: Also, do you think we can come up with a shorter name? e.g. `BetterSignedThreshold`, `SignedImproveThreshold`

File: frame/election-provider-multi-phase/src/signed.rs [hunk 3]
```diff
@@ -632,6 +632,77 @@ mod tests {
 		})
 	}
 
+	#[test]
```
Comment (kianenigma) [inline frame/election-provider-multi-phase/src/signed.rs:635]: Not a big fan of two separate tests for this. 100% and 0% are the two extremes and a bit hard to mentally follow. I suggest reading `unsigned_per_dispatch_checks_can_only_submit_threshold_better`, and creating a similar one.

**Review by kianenigma — APPROVED** [review]
LGTM other than some tests that need rework.

**Review by emostov — APPROVED** [review]
Thanks!

**Comment by kianenigma** [comment]
bot merge

**Comment by kianenigma** [comment]
\tip small

**Comment by kianenigma** [comment]
bot merge

**Comment by kianenigma** [comment]
bot merge

**Pull request merged**